### PR TITLE
Add documentation for federation overview (fixes #774)

### LINKED ISSUE #774: Consider FEDERATION.md convention
On SocialHub a sort of informal convention was adopted for describing the federation behavior of AP implementations. Currently there are about 5 (AFAIK) impls that already did so. Consider creating one as well.

Refer to: [Documenting federation behavior in a semi-standard way](https://socialhub.activitypub.rocks/t/documenting-federation-behavior-in-a-semi-standard-way/453)

I saw [Activitypub API outline](https://github.com/LemmyNet/lemmy/blob/master/docs/src/contributing_apub_api_outline.md) which is already great, and federation description will be very valuable additional info.

Particularly relevant to Lemmy is e.g. the discussion on [Groups implementation](https://socialhub.activitypub.rocks/t/groups-implementation/591) and then knowing how Lemmy's approach to Groups is. (Note: Based on that discussion you _might_ want to make changes to how it works, for compat reasons).

### PR DESCRIPTION
**Pull request by Nutomic**
Creating this PR on Github instead of Gitea as it seems relevant to a lot more people than just code. 

This is roughly based on a suggestion from the [SocialHub Forum](https://socialhub.activitypub.rocks/t/documenting-federation-behavior-in-a-semi-standard-way/453). I made some adjustments to the format to avoid duplicate content, and to better describe how Lemmy works.

This document is targeted at people that are more or less familiar with ActivityStreams and ActivityPub protocols. It is meant provide a high-level overview of ActivityPub federation in Lemmy.

### CODE REVIEW

**Review by dessalines — COMMENTED** [review]
A few things

File: docs/src/contributing_federation_overview.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+Federation
+==========
```
[the remark this answers is not in the document]
Comment (Nutomic) [reply docs/src/contributing_federation_overview.md:2]: I find this clearer in terms of readability of the plaintext, but okay I changed it.

File: docs/src/contributing_federation_overview.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+The creator of a Post, Comment or Community can delete it. It is then sent to the Community followers. The item is then hidden from all users.
+
+### Removals
+
+Mods can remove Posts and Comments from their Communities. Admins can remove any Posts or Comments on the entire site. Communities can also be removed by admins. The item is then hidden from all users.
+
+Removals are sent to all followers of the Community, so that they also take effect there. The exception is if an admin removes an item from a Community which is hosted on a different instance. In this case, the removal only takes effect locally.
+
+TODO: not sure if this accurately describes the current implementation
```
[the remark this answers is not in the document]
Comment (Nutomic) [reply docs/src/contributing_federation_overview.md:92]: If its correct then I'm removing the todo.

File: docs/src/contributing_federation_overview.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+`Delete/Note` deletes a private message.
+
+### Restore private message
+
+`Undo/Delete/Note` reverts the deletion of a private message.
+
+Community Activities
+--------------------
+
+The Community is essentially a bot, which will only do anything in reaction to actions from Users. The User who first created the Community becomes the first moderator, and can add additional moderators.
```
[the remark this answers is not in the document]
Comment (Nutomic) [reply docs/src/contributing_federation_overview.md:119]: The thing with different inboxes is something we need to fix anyway sooner or later. And it should be rather easy thanks to the refactoring. I added a link to #698 above to explain this.

**Comment by Nutomic** [comment]
Updated to address your comments, and I also rewrite the "federation philosphy" section.

I am also wondering whether we should add a more detailed explanation of how moderation works. As well as the instance blocklist/allowlist. But those might fit better into the administration docs.

Do you think its good to capitalise the terms "Community", "Comment" etc like this? And is there too much duplicate info, or any info missing?

**Comment by dessalines** [comment]
> I am also wondering whether we should add a more detailed explanation of how moderation works. As well as the instance blocklist/allowlist. But those might fit better into the administration docs.

It seems good to me, we're still kinda working out federated moderation anyway. Capitalization is fine too, especially since they're concepts. I'll merge and we can refine it later if necessary.

**Pull request merged**